src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -2899,4 +2839,220 @@ public function sanitize_post_statuses( $statuses, $request, $parameter ) {
+		$taxonomies = wp_list_filter( get_object_taxonomies( $this->post_type, 'objects' ), array( 'show_in_rest' => true ) );
+
+		if ( ! $taxonomies ) {
+			return $query_params;
+		}
+
+		$query_params['tax_relation'] = array(
+			'description' => __( 'Limit result set based on relationship between multiple taxonomies.' ),
+			'type'        => 'string',
+			'enum'        => array( 'AND', 'OR' ),
+		);
+
+		$include = array(
+			'type'    => array( 'object', 'array' ),
+			'default' => array(),
+			'oneOf'   => array(
+				array(
+					'title'       => __( 'Term ID List' ),
+					'description' => __( 'Limit result set to items with the specified terms assigned in the taxonomy.' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type' => 'integer',
+					),
+					'default'     => array(),
+				),
+				array(
+					'title'                => __( 'Term ID Taxonomy Query' ),
+					'description'          => __( 'Limit result set to items with the specified terms or their children assigned in the taxonomy.' ),
+					'type'                 => 'object',
+					'properties'           => array(
+						'terms'            => array(
+							'description' => __( 'Term IDs.' ),
+							'type'        => 'array',
+							'items'       => array(
+								'type' => 'integer',
+							),
+							'default'     => array(),
+						),
+						'include_children' => array(
+							'description' => __( 'Whether to include child terms in the terms limiting the result set.' ),
+							'type'        => 'boolean',
+							'default'     => false,
+						),
+					),
+					'additionalProperties' => false,
+				),
+			),
+		);
+		$exclude = array(
+			'type'    => array( 'object', 'array' ),
+			'default' => array(),
+			'oneOf'   => array(
+				array(
+					'title'       => __( 'Term ID List' ),
+					'description' => __( 'Limit result set to items except those with the specified terms assigned in the taxonomy.' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type' => 'integer',
+					),
+					'default'     => array(),
+				),
+				array(
+					'title'                => __( 'Term ID Taxonomy Query' ),
+					'description'          => __( 'Limit result set to items except those with the specified terms or their children assigned in the taxonomy.' ),
+					'type'                 => 'object',
+					'properties'           => array(
+						'terms'            => array(
+							'description' => __( 'Term IDs.' ),
+							'type'        => 'array',
+							'items'       => array(
+								'type' => 'integer',
+							),
+							'default'     => array(),
+						),
+						'include_children' => array(
+							'description' => __( 'Whether to include child terms in the terms excluded from the result set.' ),
+							'type'        => 'boolean',
+							'default'     => false,
+						),
+					),
+					'additionalProperties' => false,
+				),
+			),
+		);
+
+		$include_nonhier = $this->copy_hier_to_nonhier_taxonomy_schema( $include );
```

Comment:
This is more future proof if the order of the `oneOf` cases change. But if it works, I think it'd be better to simplify and just do an `unset`.

`unset( $exclude['oneOf'][1]['properties']['include_children']`.

We could do that after the `$query_params[ $base ]` and such are set. So.

```
unset( $query_params[ $base ]['oneOf'][1]['properties']['include_children'] );
unset( $query_params[ $base . '_exclude' ]['oneOf'][1]['properties']['include_children'] );
```

Reply:
Yep, that'll work.